### PR TITLE
add cypress testing library, design two e2e tests, awaiting error res…

### PR DESCRIPTION
**Pull request by jazznerd206**
What have you added/changed/fixed?

-   \*\*Added
    - LIBRARIES:
        - Cypress testing library
            - https://www.cypress.io/
            - End to end full featured testing library
            - requires `npx cypress install` (after npm i for new lint package)
            - `npx cypress open` after install will open cypress testing suite (test list and mock browser)
        - eslint-plugin-cypress
            - plugin for linting cypress library
            - .eslintrc.json file created in ./cypress
            - requires npm i` (new package added)
    - FILES 
        - /cypress folder to root
            - /integration/TSP_Tests is the location of the E2E tests
            - /support/commands is the location of the pre-test scripts
            - /fixtures && /plugins unchanged from boilerplate except for cypress handle
        - AM_PR_Testing.md pull request file (to be deleted on merge)
        - .eslintrc.json file created in ./cypress to hold config data for linter
        - cypress.json config file in root
            - sets base url to localhost:3000

-   \*\*Changed
    - FILES:
        - /components/NewResourceOverlay/index.js
            - add #newResourceButton & .open-panel to "Add new resource button" to create testing handles
            - add cypress handle to top of page
        - /components/NewResourceOverlay/overlay.js
            - add class .overlay to largest div for testing handle
            - add cypress handle to top of page
        - /components/NewResourceOverlay/form.js
            - add classes for inputs and Cancel button for testing handles
            - add cypress handle to top of page

-   \*\*Fixed
    - N/A

-   **Other information**:
    - DO NOT MERGE YET
    - error in formReducer ONLY in cypress testing environment
    - "cannot read property of 'id' from undefined
        - for some reason, the region variable that is getting passed into the reducer is coming through as undefined.
        - for all tests to pass, change stateID in constructor to null. This seems to have no effect on the data loading and also still allows the proper state to be selected from the dropdown.

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[0104/112632.596:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (Rahat-ch) [inline debug.log:1]: @jazznerd206 Is this file required to be here or is it more like a cypress out put thing? If it isn't required can we have dubug.log put into gitignore

File: cypress/integration/TSP_Tests/e2e_spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Context for E2E testing
```
Comment (Rahat-ch) [inline cypress/integration/TSP_Tests/e2e_spec.js:1]: awesome job on this love the tests you've added here

**Review by Rahat-ch — CHANGES_REQUESTED** [review]
Awesome job overall! I just have two requests before merging this in:

- Let me know if that debug.log is needed in the repo, if it isn't then let's add to gitignore
- I saw there were some classes you added which i believe is how cypress finds the elements to test. Can you create a constants folder inside of utiles so:
utils/constants
inside constants create a file called cypressClassnames.js and if you could move all the class strings from the components and the test files into there and import them where needed that would make it more scalable as we go. For example:

export const OPEN_PANEL_OVERLAY = "open panel"
export const CLOSE_PANEL_OVERLAY = "close panel"

so the convention we're following for the constants is the all caps and underscores - let me know if you need clarity on anything

File: debug.log (hunk 1 above)
Comment (jazznerd206) [reply debug.log:1]: The debug file can go into the gitignore, I'll put it in there for the following commit.

File: cypress/integration/TSP_Tests/e2e_spec.js (hunk 2 above)
Comment (jazznerd206) [reply cypress/integration/TSP_Tests/e2e_spec.js:1]: 😎

**Comment by jazznerd206** [comment]
Hey Rahat -

So here are the changes you requested. I hope I implemented the class names like you had imagined. I created a cypress class names js file to export all the strings from, and imported that file into each of the pieces of the new resource component.

I accessed each of the exports from the class names file by destructuring at the import line and using template strings for the class names in the JSX. Is this what you had in mind?

There is one conflicting file, form.js. It seems to not like the import statement, but it looks good to me. Let me know if there is anything I need to change about it.
